CMakeLists.txt

```diff
@@ -177,7 +177,7 @@ endif()
 
 # Configure Undefined Behavior Sanitizer if enabled
 if (ENABLE_UBSAN)
-  if (CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
```

Comment:
I'm not sure why this error was reported in GNU. I've tried a lot and no fixes.

https://github.com/apache/orc/actions/runs/16258126145/job/45898042155


```bash
CMake Error at /usr/local/share/cmake-3.31/Modules/FindPackageHandleStandardArgs.cmake:233 (message):
  Could NOT find Threads (missing: Threads_FOUND)
Call Stack (most recent call first):
  /usr/local/share/cmake-3.31/Modules/FindPackageHandleStandardArgs.cmake:603 (_FPHSA_FAILURE_MESSAGE)
  /usr/local/share/cmake-3.31/Modules/FindThreads.cmake:226 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  cmake_modules/ThirdpartyToolchain.cmake:38 (FIND_PACKAGE)
  CMakeLists.txt:200 (INCLUDE)
```